dialer.go

```diff
@@ -110,6 +113,10 @@ func NewDialer(ctx context.Context, opts ...DialerOption) (*Dialer, error) {
 		opt(&dialCfg)
 	}
 
+	// TODO: consider moving this registration into an option that callers control
+	if err := trace.InitMetrics(); err != nil {
+		return nil, err
+	}
```

Comment:
This code registers a number of views with opencensus. Based on my reading of the opencensus code, view registration does not include any network activity or similar work and just checks that a view name doesn't conflict with an existing registered view, in addition to saving a reference to the registered views. So, I think this is practically a zero-cost operation.

I was thinking of requiring explicit opt-in option, something like: `WithMetricsEnabled`, but because this code returns a error, I'm not sure what to do.

Reply:
I'm okay with it if we think it's near-zero. I think the caller having to configure the exporter seems like enough opt-in. 

Alternatively, an option could be achieved with a boolean that triggers this code and returns the error from the place it currently is now.

Reply:
On further thought, I think it's probably better to avoid adding another step to enabling metrics. Since our views are namespaced, the chance of a name collision also seems low.

Reply:
Given some responses to [metrics in the Go client libraries](https://github.com/googleapis/google-api-go-client/issues/388), I think we'll want to allow clients to disable metrics and tracing.

Reply:
From that issue it looks like the problem happens on import, not `InitMetrics`. is that correct?

Reply:
Ugh. Yes. So we can't do much about it for now.